Client/Application/BackgroundTask.swift

```diff
@@ -0,0 +1,40 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import UIKit
+import XCGLogger
+import Shared
+
+private let log = Logger.browserLogger
+
+// A stateful wrapper around starting/ending UIBackgroundTasks
+class BackgroundTask {
+    let application: UIApplication
+    let name: String?
+    let onExpire: () -> Void
+
+    private(set) private var taskId = UIBackgroundTaskInvalid
+
+    init(application: UIApplication, taskName: String?, onExpire: () -> Void) {
+        self.application = application
+        self.name = taskName
+        self.onExpire = onExpire
+    }
+
+    func begin() {
+        self.taskId = application.beginBackgroundTaskWithName(name) {
```

Comment:
> Actually, this probably isn't even an issue. 

Ya probably not. Just thought it was a bit odd we were capturing the value while also assigning to it. Either way, this isn't needed to fix the actual springboard crash. I'll revise the PR to just include the shutdown call when we don't have a syncable account.